### ISSUE TITLE
Issue: Stalling after Prerendering, Frozen View Counts

### ISSUE THREAD

**Issue by siavava** [L422Y/nuxt-pageviews#2]
Hi @L422Y

Thanks for creating `nuxt-pageviews`. I've just used it to integrate page views in my blog ([repo here][blog-repo], [live here][blog-live]). I am using static site generation via `nuxt generate`, and so far I've noticed two issues:

## Issue 1

Page views get frozen at the time of prerendering. I'm guessing this is because `nuxt generate` captured the static view-count from the site.

_I've tried wrapping usages of the component under `ClientOnly`, but the same result happens_.

I suspect I have to use `nuxt build` if I want to keep the views dynamic, but I figured I should check in first if you have any experiences/insights around this issue.

## Issue 2

Somehow, my deployment times went up by about 15 minutes. I checked the build logs and noticed that the deployment stalls right after prerendering all routes and spends _**exactly**_ 15 minutes there.  I searched for potential explanations and it seems the issue happens when there's a running process that nuxt has to wait for before exiting after. The same issue occurs locally via `nuxt generate`, on netlify (my blog), and on vercel (reproduction repo linked below). The delay is also always exactly 15 minutes.

## Reproduction

I've created a [minimal reproduction][minimal-repro], with three branches for three cases I've tried:

- Deploying without the views works normally. Reproduction branch [here](https://github.com/lostflux/nuxt-pageviews-minimal/tree/muted-pageviews), deployment logs [here](https://vercel.com/amittai/nuxt-pageviews-minimal/9W5gbUYnZdHQSFDRY9yPovpyAjAS).

![image](https://github.com/L422Y/nuxt-pageviews/assets/65356625/c5f125c2-dae2-4a33-8ce2-c97ec3ae1da5)

- Deploying with `nuxt-pageviews` stalls after prerendering for 15 minutes before deploying. Reproduction branch [here](https://github.com/lostflux/nuxt-pageviews-minimal), deployment logs [here](https://vercel.com/amittai/nuxt-pageviews-minimal/CCrkQmGMCGoiStowueGp9j7uvZuU).

![image](https://github.com/L422Y/nuxt-pageviews/assets/65356625/6dbc469e-5b51-40a7-80de-4bfe293ab612)

- Restricting component with `ClientOnly` &mdash; stalling still happens. Reproduction branch [here](https://github.com/lostflux/nuxt-pageviews-minimal/tree/client-only-views), deployment logs [here](https://vercel.com/amittai/nuxt-pageviews-minimal/67Q8wJ4aAghJYa8urVGaXnyXwcRq).

![image](https://github.com/L422Y/nuxt-pageviews/assets/65356625/44d88048-fb34-4460-841d-eb1b527da44d)

The same issues happen when running `nuxt generate` locally, so alternatively you can clone the [minimal reproduction][minimal-repro], check out the relevant branch, and run `npm run generate`.

---

Please let me know if you have any thoughts or workarounds on this. I can also take a stab at fixing it &mdash; I just I don't yet fully understand why the issue is happening... 🔢

Thanks for any help.

[blog-repo]: https://github.com/lostflux/blog.minimal
[blog-live]: https://amittai.space
[minimal-repro]: https://github.com/lostflux/nuxt-pageviews-minimal

**Comment by L422Y** [comment]
Hi @siavava, thanks for filing this issue!

> ## Issue 1
> Page views get frozen at the time of prerendering. I'm guessing this is because nuxt generate captured the static view-count from the site.
> I've tried wrapping usages of the component under ClientOnly, but the same result happens.
> I suspect I have to use nuxt build if I want to keep the views dynamic, but I figured I should check in first if you have any experiences/insights around this issue.

Correct re: `nuxt build` — in order to get the updated counts the backend needs to be available, see the endpoint registered here: https://github.com/L422Y/nuxt-pageviews/blob/1da5b2af42d98068475a6fbe0c0ebe8aab72fac5/src/module.ts#L70

> ## Issue 2
Re: stalling - this is a known bug in Nuxt, as a workaround you can add the following to your `nuxt.config.ts`:

```ts
...
  hooks: {
    "close": () => {
        process.exit(0)
    }
  }
...
```  

I did look into it a little, as soon as I remove the `useLazyAsyncData` call in the plugin it works as we would hope, but that kills the views data :) I also tried just using `$fetch` and got the same result.

EDIT: This seems to also be related to using a relative URL, if I swap it for an absolute URL, there is no hang.

**Comment by siavava** [comment]
Thanks for the update!

On Issue 1, I'll consider switching to `nuxt build` over `nuxt generate`, or maybe just having something ping netlify to run a build once a day.

On Issue 2, adding the hook fixed the hang on my end too &mdash; thanks for the insight!!

**Comment by L422Y** [comment]
Issue 2 was actually a `setTimeout` call in the plugin that would keep the `generate` task alive, but we don't need for that process, thanks again! :)

Resolved in v1.0.17 (https://github.com/L422Y/nuxt-pageviews/commit/c3b3c017c0bfc46bbfc4663a8ad13be5641b450c)

**Comment by siavava** [comment]
Awesome!! Thanks for the prompt action :)
